### ISSUE TITLE
ESP32 not showing serial port after Mac installation

### ISSUE THREAD

**Issue by danthegoodman1** [espressif/arduino-esp32#1084]
### Hardware:
Board:							ESP32 Dev Module
Core Installation/update date:			Yesterday
IDE name:							Arduino IDE
Flash Frequency:					N/A
Upload Speed:						N/A

### Description:
So I am using an ESP32 Dev Module, and a 2017 MBP 13" and after following the arduino installation tutorial for the ESP32 I have the boards, but the ESP32 dev module does not show up as a serial port.

Any help would be great, thanks.

**Comment by huming2207** [comment]
Have you installed the serial port driver?


Please run this command ```ls /dev | grep tty``` and paste the result here.

**Comment by danthegoodman1** [comment]
I have not installed any serial port drivers (I think?)
Here is the absurd output:
`tty
tty.Bluetooth-Incoming-Port
tty.MALS
tty.SOC
ttyp0
ttyp1
ttyp2
ttyp3
ttyp4
ttyp5
ttyp6
ttyp7
ttyp8
ttyp9
ttypa
ttypb
ttypc
ttypd
ttype
ttypf
ttyq0
ttyq1
ttyq2
ttyq3
ttyq4
ttyq5
ttyq6
ttyq7
ttyq8
ttyq9
ttyqa
ttyqb
ttyqc
ttyqd
ttyqe
ttyqf
ttyr0
ttyr1
ttyr2
ttyr3
ttyr4
ttyr5
ttyr6
ttyr7
ttyr8
ttyr9
ttyra
ttyrb
ttyrc
ttyrd
ttyre
ttyrf
ttys0
ttys000
ttys001
ttys002
ttys1
ttys2
ttys3
ttys4
ttys5
ttys6
ttys7
ttys8
ttys9
ttysa
ttysb
ttysc
ttysd
ttyse
ttysf
ttyt0
ttyt1
ttyt2
ttyt3
ttyt4
ttyt5
ttyt6
ttyt7
ttyt8
ttyt9
ttyta
ttytb
ttytc
ttytd
ttyte
ttytf
ttyu0
ttyu1
ttyu2
ttyu3
ttyu4
ttyu5
ttyu6
ttyu7
ttyu8
ttyu9
ttyua
ttyub
ttyuc
ttyud
ttyue
ttyuf
ttyv0
ttyv1
ttyv2
ttyv3
ttyv4
ttyv5
ttyv6
ttyv7
ttyv8
ttyv9
ttyva
ttyvb
ttyvc
ttyvd
ttyve
ttyvf
ttyw0
ttyw1
ttyw2
ttyw3
ttyw4
ttyw5
ttyw6
ttyw7
ttyw8
ttyw9
ttywa
ttywb
ttywc
ttywd
ttywe
ttywf`

**Comment by mistergreen** [comment]
Your board might use USB to UART. You'd need another driver for that.
https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers

**Comment by danthegoodman1** [comment]
> Your board might use USB to UART. You'd need another driver for that.
Will try that tomorrow and drop in here how it goes!

**Comment by mistergreen** [comment]
My ESP32 didn't show up on my MAC either until I installed that driver.

**Comment by danthegoodman1** [comment]
So it doesn't seem like it is installing them on my laptop. The files from the uninstall script are not there.

**Comment by richardfuchs** [comment]
> So it doesn't seem like it is installing them on my laptop. 

For MacOS X 10.9 and 10.10, see the folder called "Legacy MacVCP Driver".
It may also help if you drag&drop out the file "Silicon Labs VCP Driver.pkg" to any folder (e.g. Downloads) and start it from there

**Comment by JohnGEyles** [comment]
ESP32 crashes my OSX 10.11 machine when I try to upload my sketch (filed a separate issue awhile back, but no answers).

**Comment by mistergreen** [comment]
If your OS X is crashing, it could be because of the USB driver. My OS X crashed when I plugged in my ESP8266.
Try https://github.com/adrianmihalko/ch340g-ch34g-ch34x-mac-os-x-driver

**Comment by WeiboboHe** [comment]
Hi danthegoodman1. I almost met the same problem with ESP32. Did you figures it out now?

**Comment by JohnGEyles** [comment]
FWIW (I think it's mentioned in the other thread I started), mine worked when I used one of the USB ports on the back of my Apple monitor (a hub there, presumably) instead of the ones on the front of my Mac Pro tower.   I guess this wouldn't apply to the OP's Macbook Pro though.   Maybe try it with a powered hub ?   Try a shorter thicker cable ?   Add a battery to the ESP32 so it has plenty of power ?

**Comment by Mr-taweep** [comment]
Goog that "richardfuchs"

**Comment by bakzs** [comment]
Hi, I have the same problem too. downloaded the recommended silicon labs drivers for usb to uart but still no luck. My Mac OS version is High Sierra 10.13.5. 

Does anyone have the same problem too?

**Comment by Chris2060** [comment]
@bakzs  Yeah I do have the exact same problem on my macbook pro... I have a Esp 32 Devkit V1 and installed the driver from espressif and the recomended one above, but I still cant see it in the arduino IDE. Im sad...
Someone any idea how to fix it?

**Comment by JohnGEyles** [comment]
Try the various USB ports, try a different cable, try a USB hub ...

**Comment by Chris2060** [comment]
@JohnGEyles Thanks for your advice!

I used the port at the back right with a different usb cable after a restart of the laptop. Now it works... ;-)

**Comment by n8leon** [comment]
Installing the silabs Silicon Labs VCP Driver.pkg worked for me also. 👍 
(on OS X 10.11.6)

Seems obvious that we need the drivers, but I thought I already had them after going thru the espressif/arduino-esp32 install steps.
e.g. My "System Report / Hardware / USB" showed
   CP2102 USB to UART Bridge Controller driver

For others' future reference, you should see something like:
n8s-iMac:nleon$ ls -al /dev/tty.*
crw-rw-rw-  1 root  wheel   18,   2 Jul  4 11:14 /dev/tty.SLAB_USBtoUART

Regardless, thanks for the pointer!

**Comment by AcrimoniousMirth** [comment]
Having this issue too. 
I downloaded all the drivers I found from different sources including the ones mentioned here. It flashed up as a recognised device briefly but then vanished again and I can't reproduce the result. 
My board doesn't have a visible chip so I'm assuming its under the RF shield. Navigating to `About this Mac →  System Report →  USB` shows a _"Composite Device"_ with Product ID _"0xea60"_ by Silicon Labs so I hunted that down and its one of the **CP2102s**...

Still hunting.

**Comment by mattcony** [comment]
Anyone notice MAC drivers were updated yesterday - still no joy though

CP210x Macintosh OSX VCP Driver 5.0.5    - July 13, 2018

Known Issues and Limitations
----------------------------

	This release includes the Macintosh OSX driver for 64 bit Intel 
	Macintosh computers running MacOS X 10.11 or greater.
	For MacOS X 10.9 and 10.10, see the folder called "Legacy MacVCP Driver".

**Comment by AcrimoniousMirth** [comment]
Yeah, so I’ve tried identifying the device several other ways and it keeps showing up the same. My father’s old MacBook Pro also identifies it the same but when trying to view it in the Arduino IDE can’t find it, gets the same SOC and MALS. 
Haven’t tried a Windoze machine because don’t have one handy!

**Comment by AcrimoniousMirth** [comment]
I’m actually wondering if this is more an issue with the Arduino IDE? Anyone had any luck with other IDEs?

**Comment by mattcony** [comment]
Just caved and tried a win8.1 machine, up and running in 15min. I thought my new ESP32 board was bad, tried every usb cable in the house with no Joy on my MAC. Windows machine is fine, my MAC cannot even see the board in terminal so has to be driver related still.

**Comment by danthegoodman1** [comment]
Guys there are 2 drivers, the CP210x and CH340. I got those installed and
now all esp8266 and esp32 work with my 2017 mbp.

**Comment by kiranshashiny** [comment]
I am unable to see the SiLabs driver for the ESP32 Port  on my Mac OS - High Sierra 10.13.3 for the ESP32.
After going over others comments above - I installed both the CP2102 and the CH340 ( since I play with the Arduino and ESP8266 ) but the port for the ESP32 is not showing up.
I have multiple ports and so I plugged in the ESP8266 - and I got the "cu.wchusbserial1410" and this is good. But the /dev/cu.SLAB_USBxxx for the ESP32 is not showing !

Note : I rebooted my Mac after installing the driver.

**Comment by lionep** [comment]
I had the same issue on a Macbook pro 2017.
I was using USB-C to USB-A adaptor, and USB-A to micro usb cable.  
I added a basic usb 2.0 hub between the adaptor and the cable, and it's now detected.

**Comment by Odi55555** [comment]
Had the same issue with bei 2017 MBP. I bought a new USB A to Micro USB cable and now it works.

**Comment by JakeHartnell** [comment]
Had the same issue on a 2018 MBP, spent ages trying different cables and reinstalling drivers. On my Linux machine it just works but I have to use the MBP for work so to get it working:
- [disabled Gatekeeper](https://www.easymacsupport.com/blog/august-29th-2017) and reinstalled the driver
- Disable [System Integrity Protection](https://www.macworld.co.uk/how-to/mac/how-turn-off-mac-os-x-system-integrity-protection-rootless-3638975/)
- Disabled kext signing

Note, you may not need all those steps; I went overboard cause I am also trying to get a Linux partition working. Anyway, hope that helps someone. : )

**Comment by lbernstone** [comment]
@JakeHartnell : Seems like it would just be easier to install virtualbox and a linux vm.

**Comment by adddog** [comment]
[Silabs drivers](https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers) worked for me. Had to run the `uninstaller.sh` and then, in my case, install Legacy.

Now I see `/dev/tty.SLAB_USBtoUART` in  Arduino IDE.

**Comment by pallikas** [comment]
I had the same issue, then I realized that the usb cable i was using was set up for power only, and was not passing the signals.  I was burned by this kid of cable once before, and need to put label tags on the power only cables. Works fine, with the CP2102 Drivers and the proper cable.

**Comment by twoblokeswithapostie** [comment]
@ mistergreen Your silabs driver recommendation worked for me on High Sierra. I just had to allow loading in System Preferences and unplug and plug in the board. I can now see tty.SLAB_USBtoUART under /dev/. Thank you!

**Comment by daspuru** [comment]
I solved it this way,

1. Use terminal to go to this directory: /Library/Extensions
2. Run  sudo kextload SiLabsUSBDriver.kext/
3. If the output is an alert, go to System Preferences, and click the Allow button like this:
![68747470733a2f2f646576656c6f7065722e6170706c652e636f6d2f6c6962726172792f636f6e74656e742f746563686e6f7465732f746e323435392f4172742f746e323435395f617070726f76616c2e706e67](https://user-images.githubusercontent.com/24983936/46229386-12ba2800-c32b-11e8-920f-0e88cc410126.png)

4. Then run again the command, if nothing is in the output, you are in.
5. Use 115,200 BPS setup for programming.

**Comment by jstarr** [comment]
Had the same problem (macpro running High Sierra).  AcrimoniousMirth said we needed both CP210x and CH340 drivers.  I only found the CP210X driver on the Silab site.  Did a search for CH340 macos drivers and found https://medium.com/@tony92151/arduino-ch340-chip-driver-for-high-sierra-39e621dae1e4 which pointed to https://github.com/adrianmihalko/ch340g-ch34g-ch34x-mac-os-x-driver/raw/master/CH34x_Install_V1.3.pkg which ended up working for me.   The ls /dev/cu.* returns:
/dev/cu.usbmodem1411.

**Comment by artwebstudios** [comment]
Hi have the same issue. 

I have tried both drivers CP210x and CH340. Both of them works occasionally but after restart the computer does not work. Even when I am trying to reinstall them. It does not work. 

Any idea how to solve this issue?

**Comment by Tobils** [comment]
daspuru i have the same issue on my macbook air, with high siera. the problem was solved by click button allow on System Preferences the choose Security and Setting, then on bellow of the  page click allow. dont forget to choose . driver that you've installed. it's working for me.

**Comment by Tobils** [comment]
> Hi have the same issue.
> 
> I have tried both drivers CP210x and CH340. Both of them works occasionally but after restart the computer does not work. Even when I am trying to reinstall them. It does not work.
> 
> Any idea how to solve this issue?

try to click allow on Security & Privacy button.

**Comment by Tobils** [comment]
> I solved it this way,
> 
> 1. Use terminal to go to this directory: /Library/Extensions
> 2. Run  sudo kextload SiLabsUSBDriver.kext/
> 3. If the output is an alert, go to System Preferences, and click the Allow button like this:
>    ![68747470733a2f2f646576656c6f7065722e6170706c652e636f6d2f6c6962726172792f636f6e74656e742f746563686e6f7465732f746e323435392f4172742f746e323435395f617070726f76616c2e706e67](https://user-images.githubusercontent.com/24983936/46229386-12ba2800-c32b-11e8-920f-0e88cc410126.png)
> 4. Then run again the command, if nothing is in the output, you are in.
> 5. Use 115,200 BPS setup for programming.

 i have the same issue on my macbook air, with high siera. the problem was solved by click button allow on System Preferences the choose Security and Setting, then on bellow of the page click allow. dont forget to choose . driver that you've installed. it's working for me.

**Comment by djaramil** [comment]
I tried all kinds of things to get my ESP32 to work from here and from SLAB website to no avail.  My ESP8266 worked fine with no problem.
The one thing that eventually led me to figuring out the problem was to go to the About Mac / System Report / USB and that is when I saw that my ESP did not register under USB.  I then got another ESP32 and tried that one and it then showed up on the list and the drivers also showed up too after that.
![image](https://user-images.githubusercontent.com/6727195/50567742-96aef200-0d16-11e9-9c0a-d5daa5745b7a.png)
Sometimes it is a hardware problem.  :-(
I hope this helps.

**Comment by JohnGEyles** [comment]
Yep, I think mine may have been a bad USB cable too.

**Comment by marctheshark3** [comment]
Hello, I followed the instructions: download the two different drivers and allow in the system preferences in the security settings. What is weird is that I have a flora board that is able to connect no problem but my huzzah esp32 has no luck. So I know my ports and cable work properly. 
OS: Mojave: 10.14.2

Anything else I can try guys??

**Comment by marctheshark3** [comment]
> I tried all kinds of things to get my ESP32 to work from here and from SLAB website to no avail. My ESP8266 worked fine with no problem.
> The one thing that eventually led me to figuring out the problem was to go to the About Mac / System Report / USB and that is when I saw that my ESP did not register under USB. I then got another ESP32 and tried that one and it then showed up on the list and the drivers also showed up too after that.
> ![image](https://user-images.githubusercontent.com/6727195/50567742-96aef200-0d16-11e9-9c0a-d5daa5745b7a.png)
> Sometimes it is a hardware problem. :-(
> I hope this helps.

So my system seems to be able to find the device like above yet the arduino program isnt picking it up. So is it possible I have a bad board too??

**Comment by AhmedShaalan** [comment]
Same Problem here, My module is shown as "Composite Device" when connected via AmazonBasics USB Type C to Micro B 2.0 Cable with CP2102 Drivers installed on my MacBook Pro 2016. Although the cable is labeled as data and charging
However, it works fine with other USB-A to Micro USB.

Any idea why is that happening?

**Comment by AhmedShaalan** [comment]
> I found a thread discussing my issue with Mojave OS and recommended installing this driver: https://www.ftdichip.com/Drivers/VCP.htm
> 
> I installed it and after I was able to load my code to my huzzah32. Success! Hope this is able to help others.

It didn't work with my USB-C cable. 
Thanks anyways

**Comment by marcelodib** [comment]
- Download driver: 
    https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers
- Open zip and .dmg
- Go to legacy folder and install
- Open terminal
- Go to /Library/Extensions
- Run sudo kextload SiLabsUSBDriver.kext/

It solve the problem for me.
my board: DOIT ESP32 DEVKIT V1
MBP 2017

**Comment by mbbender** [comment]
Took me a while to read the readme... but for me I did not receive a warning when installing but I still had to execute

    To unblock, open the System Preferences Security & Privacy pane and
    unblock the system extension. See Apple Technical Note TN2459
    "User-Approved Kernel Extension Loading" for more information.

**Comment by thelinguist** [comment]
@mbbender it took me at least 5 hours of wondering why my driver wasn't working to read the readme. Mac asks for your permission to run the installer, but afterwards you must ALSO give permission for the driver to run. Once I did what it said in the readme, and opened System Preferences to give permission, it worked!

**Comment by besi** [comment]
I had some USB-C hubs that were not working. So if `/dev/cu.SLAB_USBtoUART` does not show up in your file system you might want to try another USB-C to USB hub.

**Comment by djaramil** [comment]
Also some USB cables are not compatible.  I have seen that myself. 

Also, you have to install the right version of driver for the version of Mac OS too.  Older Mac OS wants the older driver.  

Newer Mac OS new driver plus all those security pop ups, etc


Regards
David Jaramillo 
djaramil@fau.edu
561-703-6860

**Comment by oricoh** [comment]
> I solved it this way,
> 
> 1. Use terminal to go to this directory: /Library/Extensions
> 2. Run  sudo kextload SiLabsUSBDriver.kext/
> 3. If the output is an alert, go to System Preferences, and click the Allow button like this:
>    ![68747470733a2f2f646576656c6f7065722e6170706c652e636f6d2f6c6962726172792f636f6e74656e742f746563686e6f7465732f746e323435392f4172742f746e323435395f617070726f76616c2e706e67](https://user-images.githubusercontent.com/24983936/46229386-12ba2800-c32b-11e8-920f-0e88cc410126.png)
> 4. Then run again the command, if nothing is in the output, you are in.
> 5. Use 115,200 BPS setup for programming.

I wasted hours, you are the best!

**Comment by millergd** [comment]
FYI about the security pop ups, it only popped up in security preferences during the download. Once the download was over, **it disappeared**.

**Comment by flevin58** [comment]
Thank you all for the tips I found here.
Today I got my ESP32-WROOM 32 from Amazon and could not connect.

I have a Mac Book Pro Retina mid 2012 with Mojave 10.14.4

Solved the issue installing as suggested the Silicon Labs USB to UART drivers from this site:
https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers

The issue was also to choose the correct Arduino settings:

Board: ESP32 Dev Module
Upload Speed: 921600
Port: /dev/cu.SLAB_USBtoUART
Programmer AVRISP mkII

When uploading the code must press BOOT on device to avoid timeout issues

Thanks again !!!!
Fernando

**Comment by MacRaeMac** [comment]
I am using MBP Mid 2009 which using OSX Mojave, tried this driver https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers, the `/dev/cu.SLAB_USBtoUART` appear but when it upload sketch to device it failed and keep timeout.

**Comment by flevin58** [comment]
I solved that issue doing as follows:
1) first click the upload button from Arduino IDE (or VS Code PlatformIO extension)
2) you should see some dots on the terminal indicating the sw is trying to connect to device
3) press BOOT on ESP32 for a second and then release
4) you should see the upload progress on the terminal.

Note: I found out (at least with my device) that when you press BOOT you should "feel" a click on your finger, otherwise the timeout will persist.

Hope it helps.

**Comment by ankita-anand1** [comment]
I changed the USB cable and it worked

**Comment by jdieg0** [comment]
> Your board might use USB to UART. You'd need another driver for that.
> https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers

Thank you! I used to write to the board (a TTGO [Paxcounter](https://github.com/cyberman54/ESP32-Paxcounter)/`ttgov21new`) on my MacBook Pro (macOS High Sierra 10.13.6) from mid 2010. I'm not sure, how I managed to get the board running on my old laptop, but on my new MacBook Air 2018 running macOS Mojave 10.14.6 the board did not appear anymore as `/dev/tty.SLAB_USBtoUART`. Installing [this driver](https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers) did the job!

It appears under `/dev` again:

`ls -la /dev/tty*`

And uploading via Platformio works again.

```
esptool.py v2.6
Configuring upload protocol...
AVAILABLE: esp-prog, espota, esptool, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa
CURRENT: upload_protocol = esptool
Looking for upload port...
Auto-detected: /dev/cu.SLAB_USBtoUART
Uploading .pio/build/usb/firmware_ttgov21new_v1.7.841.bin
```

So just in case someone is searching for the upload error, I got this one:

```
Looking for upload port...
Error: Please specify `upload_port` for environment or use global `--upload-port` option.
For some development platforms it can be a USB flash drive (i.e. /media/<user>/<device name>)
*** [upload] Explicit exit, status 1
=============================== [ERROR] Took 9.92 seconds ===============================

======================================= [SUMMARY] =======================================
Environment ota [IGNORED]
Environment usb [FAILED]
Environment dev [IGNORED]
========================= 1 failed, 0 succeeded in 9.93 seconds =========================
The terminal process terminated with exit code: 1
```

**Comment by nalzok** [comment]
I'm using [this ESP32 development board](https://www.amazon.com/HiLetgo-ESP-WROOM-32-Development-Microcontroller-Integrated/dp/B0718T232Z), and [this USB cable](https://www.amazon.com/gp/product/B00UUBRX0Y/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1) from Amazon. After install [the driver from Silicon](https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers), I see a "Composite Device" in System Information, but `ls -la /dev/cu.*` still can't find anything new.

Any advice? I'm using macOS Mojave 10.14.6 on MacBook Pro (13-inch, 2017, Two Thunderbolt 3 ports).

<img width="996" alt="Screen Shot 2019-08-31 at 15 52 35" src="https://user-images.githubusercontent.com/13443062/64069832-5b4b2700-cc07-11e9-82a0-9ed4961bbbab.png">

**Comment by mistergreen** [comment]
try another usb port.

**Comment by nalzok** [comment]
I have two ports in total. Neither works.  :(

**Comment by ankita-anand1** [comment]
Try using a different USB cable, not all of them work...

**Comment by monomente** [comment]
For me I had to install the driver AND change to a different USB cable!

**Comment by jerik** [comment]
I had the same problem. A cable change did solve the problem. I bought an micro-usb cable that supportet datatransfer: cellularline USB Data Cable for MicroUsb Smartphone and Tables.

Some information: 
- My System: Mac OS X - 10.13.6
- Silabs Driver 5.2.2 from [Silicon Labs](https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers)
- Adafruit HUZZAH32 - ESP32 Feather Board

In the uninstaller.sh of Silicon Labs the possible location for the installation of the driver are listed. My driver was installed under `/Library/Extensions/SiLabsUSBDriver.kext`. It is an Kernel Extension which will load when you connect your adafruit board via the *correct* Micro-USB cable. 

To check if the driver loads I did the following: 
1. Getting the Kernel Extension ID

    grep -A2 undle /Library/Extensions/SiLabsUSBDriver.kext/Contents/Info.plist

For me it was `com.silabs.driver.CP210xVCPDriver`. With this information you can load the Kernel extension and check if the extension is loaded. What can be a hint, that *just* the cable is the wrong one...

    sudo kextload -v 3 -b com.silabs.driver.CP210xVCPDriver
    sudo kextstat -b com.silabs.driver.CP210xVCPDriver
    Index Refs Address            Size       Wired      Name (Version) UUID <Linked Against>
  170    0 0xffffff7f80d17000 0x9000     0x9000     com.silabs.driver.CP210xVCPDriver (5.2.2) 79BD054F-7A71-34DE-9763-59D44DAC0615 <148 22 4 3>

Good luck :)

**Comment by MadGeometer** [comment]
Am getting same problem: ESP32 not being shown in list of serial ports
- My System: macOS Catalina 10.15.2
- SiLabs CP210xVCPDriver version 5.2.3
- ESP32 board Sparkfun ESP32 Thing Plus (plus other ESP32 boards)

I've tried different USB cables, and followed the steps given by jerik above, and
ls /dev/cu*
returns only
/dev/cu.Bluetooth-Incoming-Port

This started, I think, when upgrading to Catalina. The ESP32s were listed before the upgrade. 

Any ideas??

**Comment by AndrewMoizer** [comment]
Under my old OS Mavericks (IIRC) I was never able to get the serial drivers to work. This is on a 13" MacBook Pro, late 2011 model. With 10.13.6 (High Sierra) I was able to make it go using the latest installer, being sure to approve the drivers in the Security part of the control panel (as mentioned in the installer, although the screen was different than the one shown), and doing a reboot. I can now use the Arduino tools to program the ESP32 module I got a year or so ago. Appreciate the comments here.

**Comment by mansoorahmadsamar** [comment]
> Your board might use USB to UART. You'd need another driver for that.
> https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers

This resolves my problem, Thanks

**Comment by ivancerrato** [comment]
it works with the latest driver version:
https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers
and
Arduino IDE 1.8.12 - macOS 10.15.3

crw-rw-rw- 1 root wheel 18, 7 Feb 14 08:02 cu.SLAB_USBtoUART

**Comment by m3ftah** [comment]
It used to work, but after I have updated to macOS 10.15.3, the Serial Port didn't seem to show whenever I plug my esp32.
It worked after installing the driver here https://github.com/espressif/arduino-esp32/issues/1084#issuecomment-586129694 :
https://www.silabs.com/products/development-tools/software/usb-to-uart-bridge-vcp-drivers

**Comment by andrewknockin** [comment]
Still a problem for me with a MacBook Pro running Catalina. Trying to connect the ESP32 board in Arduino development. Have downloaded the SiLabs driver and installed it. Gone through all the steps of [jerik](https://github.com/jerik) post above which is very informative.
In Arduino, when I try and select a port to connect the ESP32 the port isn't listed.
The cable works with an Arduino Uno and a Western Digital USB drive so definitely has the capability to deliver power and data.
If I go to About This Mac > System Report > Extensions I can see the driver listed but it says not loaded:
SiLabsUSBDriver:

  Version:	5.3.5
  Last Modified:	25/02/2020, 22:20
  Bundle ID:	com.silabs.driver.CP210xVCPDriver
  Notarised:	Yes
  Loaded:	No
  Get Info String:	Version 5.3.5
  Obtained from:	Identified Developer
  Kind:	Intel
  Architectures:	x86_64
  64-Bit (Intel):	Yes
  Location:	/Library/Extensions/SiLabsUSBDriver.kext
  Kext Version:	5.3.5
  Loadable:	Yes
  Dependencies:	Satisfied
  Signed by:	Developer ID Application: Silicon Laboratories Inc (52444FG85C), Developer ID Certification Authority, Apple Root CA

I run [jerik](https://github.com/jerik)'s command to load it, and it completes and says:
Requesting load of com.silabs.driver.CP210xVCPDriver.
com.silabs.driver.CP210xVCPDriver loaded successfully (or already loaded).

but the Extension list still shows it as not loaded and the Arduino app can't see it.

I've re-booted many times. Any other suggestions.

**Comment by thubold** [comment]
Had the problem that my USB-C to Micro-USB cable did not connect to the ESP32.
Turned the USB-C connector 180 degrees and it works. :-D

**Comment by matheuscouto** [comment]
Installed SiLabsUSBDriver, it shows in Extensions but wont show in Arduino port. Does it show even when the board is not connected? I'm start to thinking it's the cable.

```
SiLabsUSBDriver:

  Versão:	5.3.5
  Última Modificação:	25/02/2020 19:20
  ID do Pacote:	com.silabs.driver.CP210xVCPDriver
  Reconhecido:	Sim
  Carregada:	Sim
  String de Informações Obtidas:	Version 5.3.5
  Obtida de:	Desenvolvedor Identificado
  Tipo:	Intel
  Arquiteturas:	x86_64
  64 Bits (Intel):	Sim
  Localização:	/Library/Extensions/SiLabsUSBDriver.kext
  Versão da Ext. de Kernel:	5.3.5
  Endereço de Carregamento:	18446743521870360000
  Carregável:	Sim
  Dependências:	Satisfeitas
  Assinada por:	Developer ID Application: Silicon Laboratories Inc (52444FG85C), Developer ID Certification Authority, Apple Root CA
```

**Comment by matheuscouto** [comment]
My bad, it was the cable

**Comment by palengh** [comment]
I have simular issues. 
Running OSX 10.15.4, The Silicon Laboratories drivers are installed, and ESPHomeFlasher is working, recognizing a regular USB-connection to the ESP via an USB-C ---> USB adaptor.

Still the USB-C ---> USB-C connection does not recognize this USB-C connected ESP. 
I have tested three different cables with no luck.
I am out of ideas!

**Comment by andrewknockin** [comment]
I went all the way to a shop especially to get a new cable (before CoronaVirus). It seems older cables are not up to it. @matheuscouto above found the same thing.

**Comment by matheuscouto** [comment]
I just picked a different cable and it worked. I found out there are some cables that just charges and don’t exchange the correct data. I don’t know the spec difference between then though.

**Comment by palengh** [comment]
> I went all the way to a shop especially to get a new cable (before CoronaVirus). It seems older cables are not up to it. @matheuscouto above found the same thing.

I was at Apple Store today, and they let me test two different cables, one fresh off the shelf, but none of them worked.

**Comment by matheuscouto** [comment]
I didn’t know apple sells micro usb cables

**Comment by palengh** [comment]
> I didn’t know apple sells micro usb cables

Well, this is about USB-C ---> USB-C, which is the only connector for my ESP8266 TTGO-T Display.
[https://github.com/Xinyuan-LilyGO/TTGO-T-Display?spm=a2g0o.detail.1000023.1.59f89ca72kY3yP](url)

**Comment by palengh** [comment]
I might have found a solution to the mystery. The article "[3 problems with USB-C you need to know about](https://www.howtogeek.com/353410/3-problems-with-usb-c-you-need-to-know/)" over at howtogeek.com could explain it; 

... most of the USB-C cables on the market are USB 2.0 rather than 3.1 because they are only designed for charging. If you need to use them for anything else, like connecting devices or transferring data, they will either not work, or will be extremely slow. Make sure you’re buying the right cable. We recommend [Amazon Basics cables](https://aax-us-east.amazon-adsystem.com/x/c/QgabuQP2Qzil85Xyul80dQgAAAFxmK4AaQEAAAFKAVBhH_E/https://assoc-redirect.amazon.com/g/r/https://www.amazon.com/s/ref=nb_sb_noss?linkCode=w61&imprToken=3BKoXqqxtoTNK7OP9fa-Eg&slotNum=2&url=search-alias%3Delectronics&field-keywords=amazon+basics+usb-c+3.1&tag=823814-20).

**Comment by antonis-papaioannou** [comment]
I followed the steps described in the following link and it worked for me (max os x high sierra)!

https://github.com/adrianmihalko/ch340g-ch34g-ch34x-mac-os-x-driver

**Comment by synth** [comment]
I just pulled my hair out on this. TLDR: It was the cable :/

Here's the long version of the story I went through to get this to work if perhaps its helps people to realize that, yes, it really can be the cable! There is also additional information at the bottom regarding having a busy usb serial port error. 

So, I tried no less than 20 different micro-usb cables of varying age, length, and style (including new braided ones). Nothing worked. I installed the drivers above, restarted multiple times, nothing. 

I tried on two different machines: one running OSX 10.15.x (catalina) and the other 10.14.x (Mojave). The 10.15.x machine has only USB-c ports and I was going through a hub. The 10.14.x machine has USB-A ports. No difference between them. 

I picked up two different USB-C to USB-A adapters to rule out if the hub was an issue - no difference. 

I happened to find in a parts bin a strange cable my wife had bought for me from a kickstarter that is a USB-C to micro-usb cable and it has a magnetic end similar to Apple's magport charging adapters and that worked! 

I've never had this specific issue with arduino and micro-usb cable's port before. What a crazy thing just to get started. Thank goodness I had this weird cable. 

However, this was not the end of my saga to get the esp32 to work. Once I got the usb port to show up, trying to upload example sketches would error out with "Resource busy". For some reason the standard usb serial port kept showing up as busy and unable to connect. I tried everything that was listed in other issues and on the internet such as swapping ports, rotating the usb-c side of the cable upside down, using `lsof` to find a process that was holding the port. None of it worked. 

However, following [this comment](https://github.com/espressif/arduino-esp32/issues/1084#issuecomment-363294312),  I installed the package file that was in the `Troubleshooting` folder of the SiLabs driver which got the USB-to-UART driver installed. This port now shows in Arduino IDE and I'm able to compile and upload sketches!!

One final note for beginners is to make sure the baud rate of your serial monitor matches the baud listed in the `Serial.begin` line of your sketch :) This tripped me up, but now everything is working and I get actually begin on my project :)

**Comment by flighter18** [comment]
@synth Thank you so mutch for this "cable" hint.
I found also a cable that is working! :)

**Comment by sinkers** [comment]
@synth thanks for that, also just changed cable and device showed, fortunately for me it was the second cable I had in front of me

**Comment by yerabashtard** [comment]
It was the cable for me also!! This isn't a 100% accurate way of determining if your USB cable can send/receive data or not, but I found a hint online to look for the USB symbol on the USB A side of the cable. I found a cable with the symbol on both sides of the cable whereas the one I was using before only had it on the microUSB side and now the port is showing up!

**Comment by gramtech** [comment]
Just for the benefit of others **the cause was a USB cable** issue for me too.

I was using a cable I got with an Arduino kit so not sure why it wasn't working and spent ages assuming that its no way it could be the cable because it was meant for flashing Arduino board!!!

Was pulling my hair out at one point because I had flashed before so knew it could be drivers but still spent a while on confirming that.

Eventually after taking a step back and thinking about what all was differently the only thing I could think of was the cable so switched the cable and the device connected and worked fine!

**Comment by hadifarnoud** [comment]
in my case it was USB cable's fault.

**Comment by JohnGEyles** [comment]
> in my case it was USB cable's fault.

Lotta people (me included) having this issue.  Is there something special about these devices that makes 'em more persnickity about the USB cable used ?

**Comment by jdryfoos** [comment]
Same here. Found another USB cable and it worked. Thanks

**Comment by DareDareHub** [comment]
Thanks, Silabs driver worked for me ! 

But I had to load it after it was installed by using this terminal command :
sudo kextload /Library/Extensions/SiLabsUSBDriver.kext
https://osxdaily.com/2015/06/24/load-unload-kernel-extensions-mac-os-x/

but that command only worked if I disabled osx SIP first, see link :
https://developer.apple.com/library/archive/documentation/Security/Conceptual/System_Integrity_Protection_Guide/ConfiguringSystemIntegrityProtection/ConfiguringSystemIntegrityProtection.html#//apple_ref/doc/uid/TP40016462-CH5-SW1

**Comment by domvo** [comment]
Just for everybody out there trying to desperately make it work: it was the cable for me as well! I tried 5 different ones and it only ever started working with number 5 which was a charging cable I had lying around from my BOSE QC35 headphones. I was about to give up and get another board.

So keep trying cables! 

For the record: I did indeed have to install the Silabs USB/UART driver, so to whoever pointed this piece of the solution out and to whoever wrote the driver: thanks!

**Comment by JohnGEyles** [comment]
Mine has been working fine for awhile, since I changed cables - on a Macbook Air running High Sierra.    Soon I will be forced to upgrade it to Mojave (in order to run TurboTax 2020).   I'm scared to death things will stop working when I do that.   Can anyone shed any light on that ?

**Comment by domvo** [comment]
Working flawlessly on my Mojave MBP!

**Comment by JohnGEyles** [comment]
> Working flawlessly on my Mojave MBP!

Did you purchase it with Mojave pre-loaded, or upgrade from an earlier version ?    Nonetheless, your testimony is encouraging.

**Comment by domvo** [comment]
My Macbook Pro Late 2013 is the one running Mojave. This computer did not ship with Mojave, so it was updated. Good luck!

**Comment by randy3k** [comment]
I tried 5 different micro usb cables, only the thickest one works.

**Comment by Diamondcreeper098** [comment]
If You have tried installing the drivers and you know that your cable can transfer data then before plugging the cable press the boot button and after plugging the cable release it. I had the Same problem but doing that fixed it.

**Comment by Tobils** [comment]
> Same here. Found another USB cable and it worked. Thanks

face the same things,

**Comment by yannickh2** [comment]
Since I could not believe in an issue with my USB cable that works for everything else, I tried this link somebody posted here: https://github.com/adrianmihalko/ch340g-ch34g-ch34x-mac-os-x-driver
Not it showed up.

**Comment by pdemidyuk-ib** [comment]
In my case it was Firewall. I've turned it off on my mac and it solved the issue.

**Comment by Carle2** [comment]
Hello,
I have a mac with Big Sûr OK I don't show the port number in my IDE Arduino for ESP8266 CH340 do you know why.
I use the adaptator Apple to convert USB to micro usb c

I installed the Firmware CH340
Do you know why pleay

**Comment by justicezyx** [comment]
> Try the various USB ports, try a different cable, try a USB hub ...

This worked for me.

**Comment by dmoskovtsov** [comment]
Spent hours on that - was a cable problem at the end (had to try 8 cables to find the correct one). For those who're trying, the setup that worked for me: 

USB C 3.0 hub + USB to USB micro data **and** charging cable - 2.4 A, 24 AWG charging cable, 28 AWG data line, 1m.

**Comment by rafche** [comment]
same Issue, tried 2hours. It was the fricking CABLE

**Comment by mrprofessor** [comment]
IT WAS THE CABLE. :(

**Comment by tjsadiq** [comment]
Cable for me, glad I read through this thread!  Running a MacBook Pro (13-inch, 2017, Two Thunderbolt 3 ports) and 11.4.   My other cables work with other boards (Adafruit M0, MKR 1010) so I was shocked.  Did the driver updates to no avail.  Then used a 12" NIMASO USB-c to USB-C with a ARKTEK adapter and it worked, showing up as /dev/cu.usbserial-0001.  Still wouldn't load the sketch--had to hold the Boot button to get the sketch to load.  Hope it helps someone!

**Comment by jginsburgn** [comment]
> I had the same issue on a Macbook pro 2017.
> I was using USB-C to USB-A adaptor, and USB-A to micro usb cable.
> I added a basic usb 2.0 hub between the adaptor and the cable, and it's now detected.

That works!!!!!!!!!!!

![IMG_4109](https://user-images.githubusercontent.com/10511301/133197211-49452b4b-dfe7-4ea3-b24d-3642698f1abc.JPG)

**Comment by Boxolymp** [comment]
aaand still, the CABLE

**Comment by Carle2** [comment]
Hi do you have a link to buy this cable for me not work and I don't want to buy 1000 cables ;)
 
 
  Le lun., déc. 6, 2021 à 17:27, ***@***.***> a écrit:   


aaand still, the CABLE

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Boxolymp** [comment]
Hey there, i used a usbc to usb adapter and a mini usb to usb cable. I just an very cheap usb mini - usb cable. Just try a bit better cable…. It’ll work iiif your set up is fine.

Best greatings

Charlene

**Comment by jginsburgn** [comment]
> Hi do you have a link to buy this cable for me not work and I don't want to buy 1000 cables ;)
I'd recommend that you make sure that the cable you buy can transfer data. There are certain cables that can only charge.

**Comment by jerik** [comment]
For me cellularline USB Data Cable for MicroUsb worked. See https://github.com/espressif/arduino-esp32/issues/1084#issuecomment-541208929[comment](https://github.com/espressif/arduino-esp32/issues/1084#issuecomment-541208929)

**Comment by Carle2** [comment]
Hello
Finally I bought a cable micro usb to usb-c and it's work :)I bought it in eBay Thank you for your help 
 
  Le mer., déc. 8, 2021 à 23:06, J. ***@***.***> a écrit:   


For me cellularline USB Data Cable for MicroUsb worked. See https://github.com/espressif/arduino-esp32/issues/1084#issuecomment-541208929[comment](https://github.com/espressif/arduino-esp32/issues/1084#issuecomment-541208929)

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by mahboud** [comment]
I am having similar problems too. I've ruled out the cable as I can use the cable for other data tasks.

I think it is my Big Sur Mac, and this is why I think so:

```
~$ ls -al /dev/tty.*
crw-rw-rw-  1 root  wheel  0x25000000 Dec 17 02:26 /dev/tty.Bluetooth-Incoming-Port
~$ cd /Library/Extensions
Extensions$ ls
CP210xVCPDriver.app  HighPointIOP.kext    HoRNDIS.kext         SoftRAID.kext        intelhaxm.zip
Dropbox.kext         HighPointRR.kext     SiLabsUSBDriver.kext intelhaxm.kext
Extensions$ sudo kextload SiLabsUSBDriver.kext/
Password:
Executing: /usr/bin/kmutil load -p /Library/Extensions/SiLabsUSBDriver.kext
Error Domain=KMErrorDomain Code=71 "Unsupported Error: one or more extensions are unsupported to load 	Kext com.silabs.driver.CP210xVCPDriver v6.0.2 in executable kext bundle com.silabs.driver.CP210xVCPDriver at /Library/StagedExtensions/Library/Extensions/SiLabsUSBDriver.kext
Unsupported Error: one or more extensions are unsupported to load 	Kext com.silabs.driver.CP210xVCPDriver v6.0.2 in executable kext bundle com.silabs.driver.CP210xVCPDriver at /Library/StagedExtensions/Library/Extensions/SiLabsUSBDriver.kext" UserInfo={NSLocalizedDescription=Unsupported Error: one or more extensions are unsupported to load 	Kext com.silabs.driver.CP210xVCPDriver v6.0.2 in executable kext bundle com.silabs.driver.CP210xVCPDriver at /Library/StagedExtensions/Library/Extensions/SiLabsUSBDriver.kext
Unsupported Error: one or more extensions are unsupported to load 	Kext com.silabs.driver.CP210xVCPDriver v6.0.2 in executable kext bundle com.silabs.driver.CP210xVCPDriver at /Library/StagedExtensions/Library/Extensions/SiLabsUSBDriver.kext}
```

Any ideas what to do? BTW, I don't get the option to "Allow" this in the Security and Privacy system prefs.

**Comment by VojtechBartoska** [comment]
Hello, I'm closing this issue as a problem was in cable. Marking with label "for Reference" and we will consider adding this later into our FAQ section in the documentation.

**Comment by kragrrr** [comment]
Is there a possible solution for Apple Silicon Macs in regards to this?

**Comment by jankammerath** [comment]
> Is there a possible solution for Apple Silicon Macs in regards to this?

I have an M1 Pro MBP 14" 2021 with Ventura 13.1. Had the same issues as described here. The following worked for me:

1. Installed the Silicon Legacy Driver
2. Used a proper MicroUSB to USB-A cable (had issues with a cheapo one)
3. Connected using a USB hub (had issues with the Apple adapter)

**Comment by arthurnn** [comment]
indeed ! changing the cable seal the deal for me

**Comment by konifer44** [comment]
Changing cable also helped 😊

**Comment by Orf1** [comment]
> > Is there a possible solution for Apple Silicon Macs in regards to this?
> 
> I have an M1 Pro MBP 14" 2021 with Ventura 13.1. Had the same issues as described here. The following worked for me:
> 
> 1. Installed the Silicon Legacy Driver
> 2. Used a proper MicroUSB to USB-A cable (had issues with a cheapo one)
> 3. Connected using a USB hub (had issues with the Apple adapter)

Hey, is there any chance you could drop a link to the "Silicon Legacy Driver" you used? Having the same issue.

**Comment by baharalidurrani** [comment]
CH340C does not work using a Thunderbolt port you've to use a USB 2.0 hub

**Comment by Upskxkit** [comment]
Hi, I have Mac OS "Sonoma" 14.0 and it was a cable.

**Comment by bluemellophone** [comment]
I was able to connect to a KeeYees ESP32 with `esptool.py` on a Apple M1 Max MacBook Pro running macOS Sonoma (14.1.1):

1. Download the CP210x drivers here: https://www.silabs.com/developers/usb-to-uart-bridge-vcp-drivers?tab=downloads
2. Install the standard driver, no need to use the legacy driver
3. Approve the System Extension in `Settings > Privacy & Security`
4. Reboot
5. Verify driver system extension is running with `systemextensionsctl list` and look for `com.silabs.cp210x	[activated enabled]`
6. Connect the ESP32 to a proper micro USB to USB-A cable
7. Connect the USB-A cable into an Apple USB-C dongle (e.g., https://support.apple.com/en-us/HT207807)
8. Verify ESP32 TTY with `ls /dev/tty.* | grep UART`
9. Verify ESP32 is up and running with `esptool.py read_mac`

You should see something like this:

```
$ esptool.py read_mac
esptool.py v4.6.2
Found 3 serial ports
Serial port /dev/cu.usbserial-0001
Connecting......
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting....
Detecting chip type... ESP32
Chip is ESP32-D0WD-V3 (revision v3.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: XX:XX:XX:XX:XX:XX
Uploading stub...
Running stub...
Stub running...
Hard resetting via RTS pin...
```

**Comment by impigra** [comment]
Another issue I figured out is that there is a USB-C style connection port but only with USB-A support. There is no resistor or id chip behind the port. With an adapter from USB-A to USB-C cable it works.

**Comment by unlucio** [comment]
I failed at step 8 of @bluemellophone's list :(
no uarts for me

**Comment by bluemellophone** [comment]
I had the same issue, but I used a different dongle / USB cable and showed right up.

**Comment by arivvid27** [comment]
I found that the issue is usually with the cable, or drivers not being installed. Once you install the drivers and the ESP32 is still in appearing the the serial ports, it is probably you are using a power only cable, instead of a data+power cable.

**Comment by Goldboter** [comment]
I had the same problem. Read this thread a year ago for the first time. Tried the driver. Didn't work.
Then I tried just a day ago different cables USB A to micro USB cables and one of them worked... I'm more than happy :D

**Comment by Parhosha** [comment]
CABEL

**Comment by hoopengo** [comment]
tbh it's not working for me in 3 another pc (linux laptop, macos laptop, windows pc), but it's finally works on mac when i use adapter type-c to usb3.0 and another usb to micro-usb cabel. It's so awful..

**Comment by cris-maillo** [comment]
> Another issue I figured out is that there is a USB-C style connection port but only with USB-A support. There is no resistor or id chip behind the port. With an adapter from USB-A to USB-C cable it works.

this finally worked! I tried everything else (multiple usb-c data cables) but it wasnt until I had USB A to an adapter that it worked (I have a Macbook air M1 and ESP32-WROOM-32 with the CP210x USB to UART Bridge drive

**Comment by Tripp-Sautner** [comment]
I have a ton of diffrent wires and ONE worked. All of these were data cables but only one worked. Absolutely bonkers

**Comment by geohndz** [comment]
I was getting exit code 2 and changing the upload speed from 921600 to 115200 worked!

You can change it in the Tools dropdown.

**Comment by mohammad-aqajani** [comment]
i'v installed multiple drivers: 

![Image](https://github.com/user-attachments/assets/437caeb5-0305-4896-a0f9-23fca743b6fb)

disable the SIP in OS recovery mode

test the Windows on Parallels virtual machine

tried 3 cables on different ports

and none of them worked yet!

**Comment by mistergreen** [comment]
Look inside the micro usb port. You should see 6 small pins. This is a data cable and not power cable.

**Comment by xwyattyx** [comment]
The silicon driver won't even install on my intel mac. I've been trying for 8 hours, to no avail. Horrible software service, should've known based on the 2009 website UI. Always a good sign when the website is horribly unusable and solid colors are pixelated. Everyone says cable issue but my driver won't even install lmfao. My 4 year old mac is basically obsolete when it comes to such simple tasks, its taken me hours to trouble shoot this just to upload some code to a $3 microcontroller thats going to take me most likely less than 30 mins to do lol.

**Comment by Matheo5** [comment]
Hi ! 
I got the same problem with my esp32 (firebeetle from DFRobot using usb-c and USB to UART module). 
I have a MacBook Pro from 2013, on macOS BigSur 11.7.10 a find out I needed a driver to use Arduino IDLE ;) 
The driver from slabs CP210x didn't worked for me after many try. 
So I find a GitHub page (https://github.com/WCHSoftGroup/ch34xser_macos) and download the CH340 driver following the instructions. 
Wording for me now ! 
Hope I'll help some ;)

**Comment by armchairdeity** [comment]
I had to use a m-m USB-C data cable... using a USB-A to USB-C did not work.

Also haven't use the magic sequence of plug it in, hold the boot button, slow-click the reset button, then release the boot button. Then even Arduino IDE recognized it.
